### DISCUSSION TITLE
tera shows the actual template code

### DISCUSSION THREAD

**Discussion by JLT032** [rwf2/Rocket#2589] [Questions]
as a first i'm trying to get a simple one page site to work with tera templates
it works, but not as it shows the actual html code generated by the templates

this works by loading mindex.tera as mindex
mindex loads a header, nav, footer which all contains pure html and a few {{ fields }}

there is no error code displayed, all seems to work well

**Comment by robertwayne** [comment]
It sounds like it's being served as `text/plain` instead of `text/html`.

Are you calling the [Template::render fn](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/index.html)? Can you share the code for your route handler?

**Reply by JLT032** [reply]
Problem is indeed rendering as text/plain.

Just found it is because i used templatename.tera instead of templatename.html.tera
